### ISSUE TITLE
Download and Parse Public Data Released by B3 Exchange

### ISSUE THREAD

**Issue by msperlin** [ropensci/software-review#534]
**Date accepted:** 2023-02-08

Submitting Author Name: Marcelo S. Perlin
Submitting Author Github Handle: @msperlin
Other Package Authors Github handles: (comma separated, delete if none) @wilsonfreitas
Repository:  https://github.com/wilsonfreitas/rb3
Version submitted: 0.0.3
Submission type: Standard
Editor: @emilyriederer
Reviewers: @pachadotdev, @quishqa
Due date for @pachadotdev: 2022-10-16
Due date for @quishqa: 2022-10-24
Archive: TBD
Version accepted: TBD
Language: en
---

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: rb3
Title: Download and Parse Public Data Released by B3 Exchange
Description: Download and parse public files released by B3 and convert them
    into useful formats and data structures common to data analysis
    practitioners.
Version: 0.0.3
Authors@R: c(person("Wilson", "Freitas",
                    email = "wilson.freitas@gmail.com",
                    role = c("aut", "cre")),
             person("Marcelo", "Perlin",
                    email = "marceloperlin@gmail.com",
                    role = "aut"))
License: MIT + file LICENSE
LazyData: true
Depends:
    R (>= 4.1.0),
Imports:
    bizdays,
    stringr,
    proto,
    cli,
    readr,
    dplyr,
    rvest,
    httr,
    jsonlite,
    purrr,
    ascii,
    rlang,
    methods,
    yaml,
    digest,
    base64enc
Suggests:
    testthat,
    knitr,
    DT,
    miniUI,
    shiny,
    xtable,
    rmarkdown,
    ggplot2,
    covr,
    scales,
    magrittr,
    tibble,
    tidyr,
    fixedincome
Collate:
    'rb3-package.R'
    'util.R'
    'transmute.R'
    'fields.R'
    'handlers.R'
    'marketdata.R'
    'download-data.R'
    'file.R'
    'convert_to.R'
    'scraper-cdi.R'
    'scraper-futures.R'
    'scraper-yc.R'
    'scraper-cotahist.R'
    'addin-show-templates.R'
    'addin-display-template.R'
    'readers.R'
    'downloaders.R'
    'zzz.R'
BugReports: https://github.com/wilsonfreitas/rb3/issues
URL: https://github.com/wilsonfreitas/rb3,
    http://wilsonfreitas.github.io/rb3/
VignetteBuilder: knitr
Roxygen: list(markdown = TRUE)
RoxygenNote: 7.1.2
Config/testthat/edition: 3
Encoding: UTF-8
```

## Scope

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) this package falls under: (Please check an appropriate box below. If you are unsure, we suggest you make a pre-submission inquiry.):

	- [X] data retrieval
	- [X] data extraction
	- [ ] data munging
	- [ ] data deposition
    - [ ] data validation and testing
	- [ ] workflow automation
	- [ ] version control
	- [ ] citation management and bibliometrics
	- [ ] scientific software wrappers
	- [ ] field and lab reproducibility tools
	- [ ] database software bindings
	- [ ] geospatial data
	- [ ] text analysis

- Explain how and why the package falls under these categories (briefly, 1-2 sentences):

The package downloads and organizes raw financial data directly from B3, the main financial exchange in Brazil. These datasets are not available in any other way.

-   Who is the target audience and what are scientific applications of this package?

Academic researchers and practioners of financial markets.

-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?

No.

-   (If applicable) Does your package comply with our [guidance around _Ethics, Data Privacy and Human Subjects Research_](https://devguide.ropensci.org/policies.html#ethics-data-privacy-and-human-subjects-research)?

Yes.

-   If you made a pre-submission inquiry, please paste the link to the corresponding issue, forum post, or other discussion, or @tag the editor you contacted.

-   Explain reasons for any [`pkgcheck` items](https://docs.ropensci.org/pkgcheck/) which your package is unable to pass.

## Technical checks

Confirm each of the following by checking the box.

- [X] I have read the [rOpenSci packaging guide](https://devguide.ropensci.org/building.html).
- [X] I have read the [author guide](https://devdevguide.netlify.app/authors-guide.html) and I expect to maintain this package for at least 2 years or to find a replacement.

This package:

- [X] does not violate the Terms of Service of any service it interacts with.
- [X] has a CRAN and OSI accepted license.
- [X] contains a [README with instructions for installing the development version](https://ropensci.github.io/dev_guide/building.html#readme).
- [X] includes [documentation with examples for all functions, created with roxygen2](https://ropensci.github.io/dev_guide/building.html#documentation).
- [X] contains a vignette with examples of its essential functions and uses.
- [X] has a [test suite](https://ropensci.github.io/dev_guide/building.html#testing).
- [X] has [continuous integration](https://ropensci.github.io/dev_guide/ci.html), including reporting of test coverage using services such as Travis CI, Coveralls and/or CodeCov.

## Publication options

- [X] Do you intend for this package to go on CRAN?
- [ ] Do you intend for this package to go on Bioconductor?

- [ ] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:

<details>
<summary>MEE Options</summary>

- [ ] The package is novel and will be of interest to the broad readership of the journal.
- [ ] The manuscript describing the package is no longer than 3000 words.
- [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
- (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no guarantee that your manuscript will be within MEE scope.*)
- (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
- (*Please do not submit your package separately to Methods in Ecology and Evolution*)

</details>

## Code of conduct

- [X] I agree to abide by [rOpenSci's Code of Conduct](https://ropensci.org/code-of-conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by msperlin** [comment]
@wilsonfreitas

**Comment by mpadge** [comment]
@msperlin The build failure in the checks above is because
```
Error: 'load_builtin_calendars' is not an exported object from 'namespace:bizdays'
```
That generally means you're not locally using latest versions of all dependency packages. Please let us know when you've fixed that, and the other issues identified above. Thanks.

**Comment by emilyriederer** [comment]
Hello @msperlin and many thanks for your submission.

We are discussing whether the package is in scope and need a bit more information.

I see you are the author of other packages such as [GetDFDData2](https://cran.r-project.org/web/packages/GetDFPData2/index.html) already on CRAN. Could you please help to clarify the differences between these two packages? Is the former focused only on financial statements whereas this package provides more granular time series? While I'm sure your two packages are not "competing", I want to fully understand what makes this package novel and [how it is best in class](https://devguide.ropensci.org/policies.html#overlap).

Thank you!

**Comment by msperlin** [comment]
Hi @emilyriederer 

Please find my replies below:

> I see you are the author of other packages such as [GetDFDData2](https://cran.r-project.org/web/packages/GetDFPData2/index.html) already on CRAN. Could you please help to clarify the differences between these two packages? Is the former focused only on financial statements whereas this package provides more granular time series? While I'm sure your two packages are not "competing", I want to fully understand what makes this package novel and [how it is best in class](https://devguide.ropensci.org/policies.html#overlap).

The different between GetDFPDAta2 and rb3 is in their scope and audience. Package GetDFPData2 is related to financial documents released by companies  to the exchange and its audience is mostly business analysts. This includes sales, profit, and many other financial information on a annual basis. Meanwhile, you cant think of rb3 as an interface to all **trade** data available at the b3 website, including prices of many different type of contracts such as options, equities (companies) and futures. ITs audience is more related to traders and market participants.

While there is some relationship between datasets from rb3 (stock prices) and GetDFPData2 (financial statements) for equities, there is none for other types of markets.

> Similarly, I see a B3 data listed on general services like [Yahoo Finance](https://finance.yahoo.com/quote/B3SA3.SA/) which I know can be accessed through a number of packages including `yfR`, `quantmod`, and `tidyquant`. Could you elaborate on why none of these packages are in a "comparison set"?
 
The data is indeed simillar, but only for equities (companies). Yahoo finance, for example, does not provide historical prices for futures or option contracts. Package rb3 also provides access to historical yield curves, which is a very unique (and rare) set of data in finance.

**Comment by msperlin** [comment]
@wilsonfreitas I just fixed the code for codemeta and contributing file. The other problems are deeper in the code. Can you please have a look?

**Comment by wilsonfreitas** [comment]
Hi @emilyriederer and @msperlin

I added the examples to the functions show_templates and display_templates.
rb3 is not a function, it is just the tag `@name` being used in the package documentantion, I removed that.

And with respect to `<<-` operator, it is used with a ReferenceClass, and accoding to [documentation](https://www.rdocumentation.org/packages/methods/versions/3.6.2/topics/ReferenceClasses) this is the suggest way to modify objects fields.
So, I believe that this is not an issue.

**Comment by wilsonfreitas** [comment]
Furtherly, the test coverage is now 84%.

**Comment by mpadge** [comment]
Thanks @wilsonfreitas, our check system has just been updated to do what above link says: *Allow global assignment operator if in RefClass statement*. Your package no longer fails that check :+1:

And FYI, another new check has just been added which in your case reports this:

---

### 4. Other Checks

:heavy_multiplication_x: The following function name is duplicated in other packages:

-  - `download_data` from admixr, FedData, fgeo.x, fHMM, hkdatasets, pRecipe, Rnumerai

---

That's currently a "trial" function so you don't necessarily need to respond, but feel free to change the name if you want.

**Comment by wilsonfreitas** [comment]
Hi @mpadge

> Thanks @wilsonfreitas, our check system has just been updated to do what above link says: Allow global assignment operator if in RefClass statement. Your package no longer fails that check 👍

This is great!

---

* `download_data` has been renamed to `download_marketdata`.

**Comment by emilyriederer** [comment]
@ropensci-review-bot check package

**Comment by emilyriederer** [comment]
Thanks for the answers on the package uniqueness, @msperlin . I really appreciate the additional context on the package's unique value. I will proceed to look for editors for this package. Would you mind adding more details and context to the README? After reading it someone with little domain knowledge should have been informed about the aim, goals and functionality of the package.

**Comment by msperlin** [comment]
Thanks @emilyriederer. I added to the README.md the explicit datasets available at rb3. This should give context to the reader and set rb3 apart from other packages.

**Comment by emilyriederer** [comment]
Hello again, @msperlin ! 

Thanks for updating the README. 

I noticed that you are also currently going through active review for `yfR` (#523) right now. Discussing with the editorial board, we think the best policy is for developers to undertake reviews sequentially. That way, any relevant feedback or discussions from one review can be applied more seamlessly to the next. 

As such, I am going to apply the `hold` tag for now. Once #523 is complete, please ping me in this thread and we will pick back up where we left off.

**Comment by maelle** [comment]
@ropensci-review-bot put on hold

**Comment by maelle** [comment]
@msperlin do you still intend to submit this package?

**Comment by msperlin** [comment]
Hi @maelle, 

@wilsonfreitas is the main author of the package and has done far more work on the code than myself. I believe the decision should be his.

**Comment by msperlin** [comment]
Great! I believe the code should be fine. I'll try to start the check from here.

**Comment by msperlin** [comment]
@ropensci-review-bot check package

**Comment by mpadge** [comment]
@msperlin and @emilyriederer you may ignore the "functions do not document return values" fail, as that has only come online long after this package was submitted. @msperlin you may of course update your code to document these, as they'll also fail CRAN checks for new packages, but that's up to you. The duplicated names checks may also be ingored for the moment.

**Comment by emilyriederer** [comment]
Thanks for the clarification, @mpadge ! With that, I think it sounds like we should be ready to proceed with the review. I will begin to work towards assigning a handling editor.

**Comment by emilyriederer** [comment]
@ropensci-review-bot assign @maurolepore as editor

**Comment by emilyriederer** [comment]
@ropensci-review-bot assign @emilyriederer as editor

**Comment by emilyriederer** [comment]
Hi @msperlin and @wilsonfreitas - since my EiC term has ended, I have self-volunteered to be the editor for this package as well. I will begin considering reviewers, but in the mean time, please let me direct your attention to a few items found in the automated checks:

**Comment by emilyriederer** [comment]
### Editor checks:

- [x] **Documentation**: The package has sufficient documentation available online (README, pkgdown docs) to allow for an assessment of functionality and scope without installing the package. In particular,
    - [x] Is the case for the package well made?
    - [x] Is the reference index page clear (grouped by topic if necessary)?
    - [x] Are vignettes readable, sufficiently detailed and not just perfunctory?
- [x] **Fit**: The package meets criteria for [fit](https://devguide.ropensci.org/policies.html#package-categories) and [overlap](https://devguide.ropensci.org/policies.html#overlap).
- [x] **Installation instructions:** Are installation instructions clear enough for human users?
- [ ] **Tests**: If the package has some interactivity / HTTP / plot production etc. are the tests using [state-of-the-art tooling](https://devguide.ropensci.org/building.html#testing)?
- [x] **Contributing information**: Is the documentation for contribution clear enough e.g. tokens for tests, playgrounds?
- [x] **License:** The package has a CRAN or OSI accepted license.
- [x] **Project management**: Are the issue and PR trackers in a good shape, e.g. are there outstanding bugs, is it clear when feature requests are meant to be tackled?
---

#### Editor comments

I think this package is quite close to being read for review. I just want to note a few core issues:

- Since this is an API-heavy package, please consider the [rOpenSci guidance](https://books.ropensci.org/http-testing/) for http testing best practices. I think some of the tools listed there could be helpful to make the current checks more robust
- In the `CONTRIBUTING.md`, I notice there are no instructions about setting up API keys for testing. Can you please confirm that users do not need to provide any authentication information for these APIs? 

Additionally, from the automated checks, I want to highlight a few items in particular:

- Some stated imports (e.g. `cli`) are not found as being used. Can this dependency be removed?
- Please review and resolve the formatting issues found by `lintr`
- You may consider whether it is a concern that `index_get()` is a function name used in other packages. I doubt if this package would be used much with `elastic` so I don't foresee the conflict arising frequently, but if another name suits you equally well, it might be better to change

---

**Comment by wilsonfreitas** [comment]
Hi @emilyriederer,

Thanks for your suggestions.
I'm answering your questions here.

> Tests: If the package has some interactivity / HTTP / plot production etc. are the tests using [state-of-the-art tooling](https://devguide.ropensci.org/building.html#testing)?
Ok, I see valuable points here, I'll work on them.

> Project management: Are the issue and PR trackers in a good shape, e.g. are there outstanding bugs, is it clear when feature requests are meant to be tackled?
Yes, it is, but we don't have an structured process to be managed.

I'm going to work on the other points: dependencies, formatting, function names ...
Thanks

**Comment by emilyriederer** [comment]
Hi @wilsonfreitas ! I hope you are well. I thought I'd follow up and see if you had a chance to make any updates in response to my comments? Most all of my comments were a bit open-ended and not a hard, binary requirement. Once you've incorporated them to your satisfaction, I can begin to look for reviewers.

**Comment by wilsonfreitas** [comment]
Hi @emilyriederer, thanks!!

Unfortunately I am slower than I'd like, but I have a few updates.

> In the CONTRIBUTING.md, I notice there are no instructions about setting up API keys for testing. Can you please confirm that users do not need to provide any authentication information for these APIs?

The package makes heavy use of web scraping, it does not require any API to access the data and no setup is necessary for testing.
Should I make this clear in the CONTRIBUTING.md?

> Please review and resolve the formatting issues found by lintr

Done!

> You may consider whether it is a concern that index_get() is a function name used in other packages. I doubt if this package would be used much with elastic so I don't foresee the conflict arising frequently, but if another name suits you equally well, it might be better to change

If this is not blocking I'd prefer to keep `index_get` as function name.

> Some stated imports (e.g. cli) are not found as being used. Can this dependency be removed?

Currently, I am working on this point.

> Since this is an API-heavy package, please consider the [rOpenSci guidance](https://books.ropensci.org/http-testing/) for http testing best practices. I think some of the tools listed there could be helpful to make the current checks more robust

I am reading this book to see what can be applied in the package.

**Comment by wilsonfreitas** [comment]
Hi @emilyriederer 

I think I am done for now.
The last commits have many changes implementing your suggestions.
We can go forward with the process.

**Comment by emilyriederer** [comment]
@ropensci-review-bot check package

**Comment by emilyriederer** [comment]
Thanks @wilsonfreitas ! This all looks great. I will begin to look for reviewers 🎉

**Comment by emilyriederer** [comment]
@ropensci-review-bot seeking reviewers

**Comment by emilyriederer** [comment]
@ropensci-review-bot assign @pachadotdev as reviewer

**Comment by emilyriederer** [comment]
@ropensci-review-bot assign @quishqa as reviewer

**Comment by emilyriederer** [comment]
@wilsonfreitas , please meet your reviewers @quishqa and @pachadotdev . I am particularly excited to have them reviewing this package as they are both rOpenSci package developers themselves with packages that support different data extraction tasks focused on South America. Lots of commonalities and opportunities to share best practices here!

**Comment by emilyriederer** [comment]
@pachadotdev and @quishqa - friendly reminder on this review!

**Comment by pachadotdev** [comment]
hi!

just an initial round of questions

1. can you please add vcr to suggests? I would prefer to be able to test offline without skipping the test but having the cassettes to run the test anyways, as in https://github.com/ropensci/tradestatistics/blob/master/tests/testthat/test-ots_create_tidy_data.R#L9

2. do you have a roadmap to test the API-centric functions? these seem to be the most important but here only 35% of code lines are covered by test cases.

**Comment by emilyriederer** [comment]
Hi @pachadotdev ! Thanks for starting to dog into the review. Could you please use our review template [here](https://devguide.ropensci.org/reviewtemplate.html#reviewtemplate) to collect your questions and comments? I definitely appreciate the push for testing best practices, but we also want to make sure authors can get the "big picture ideas" out of the reviews

**Comment by emilyriederer** [comment]
@pachadotdev and @quishqa - I know it's a busy time of the academic year, but since we are at/over the end of the review periods, could you please provide an update on your availability and when you might be able to complete reviews? Thank you!

**Comment by pachadotdev** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below.  Your review is not limited to these topics, as described in the reviewer guide*

- **Briefly describe any working relationship you have (had) with the package authors.**
- [x] As the reviewer I confirm that there are no [conflicts of interest](https://devguide.ropensci.org/policies.html#coi) for me to review this work (if you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need:** clearly stating problems the software is designed to solve and its target audience in README
- [x] **Installation instructions:** for the development version of package and any non-standard dependencies in README
- [x] **Vignette(s):** demonstrating major functionality that runs successfully locally
- [x] **Function Documentation:** for all exported functions
- [x] **Examples:** (that run successfully locally) for all exported functions
- [x] **Community guidelines:** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [ ] **Automated tests:** Unit tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines.

Estimated hours spent reviewing: 3

- [x] Should the author(s) deem it appropriate, I agree to be acknowledged as a package reviewer ("rev" role) in the package DESCRIPTION file.

---

### Review Comments

1. can you please add vcr to suggests? I would prefer to be able to test offline without skipping the test but having the cassettes to run the test anyways, as in https://github.com/ropensci/tradestatistics/blob/master/tests/testthat/test-ots_create_tidy_data.R#L9

2. do you have a roadmap to test the API-centric functions? these seem to be the most important but here only 35% of code lines are covered by test cases.

1+2 would really simplify the testing for this package

overall it looks good, my only criticism is the lack of a wide battery of tests

**Comment by wilsonfreitas** [comment]
Hi @pachadotdev 

Thanks for your review.
vcr has been added to Suggests.
I believe I don't have a structured roadmap to test the web-scraping functions.
I understand and agree with your suggestions and I'd love to put more work on that, but at this moment I am struggling out of time.

**Comment by quishqa** [comment]
Hi all! 
Sorry for the delay, I'll send my review at the end of the week!

**Comment by quishqa** [comment]
Dear @emilyriederer , and @wilsonfreitas , sorry once again for the delay. 
Here are my comments.

## Package Review

*Please check off boxes as applicable, and elaborate in comments below.  Your review is not limited to these topics, as described in the reviewer guide*

- **Briefly describe any working relationship you have (had) with the package authors.**
- [x] As the reviewer I confirm that there are no [conflicts of interest](https://devguide.ropensci.org/policies.html#coi) for me to review this work (if you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need:** clearly stating problems the software is designed to solve and its target audience in README
- [x] **Installation instructions:** for the development version of package and any non-standard dependencies in README
- [x] **Vignette(s):** demonstrating major functionality that runs successfully locally
- [x] **Function Documentation:** for all exported functions
- [x] **Examples:** (that run successfully locally) for all exported functions
- [x] **Community guidelines:** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [ ] **Automated tests:** Unit tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines.

Estimated hours spent reviewing: 6

- [x] Should the author(s) deem it appropriate, I agree to be acknowledged as a package reviewer ("rev" role) in the package DESCRIPTION file.

---

### Review Comments

`rb3` is a solid package that works and fulfills its objectives. I ran the examples in the README and the vignettes without problems. 
I used  `pkgreviewr` to help me review `rb3`. The complete report is [here](https://github.com/quishqa/my_ropensci_reviews/blob/main/rb3-review/rb3-review.md). These are my main comments:

- When running  `goodpractice`, It still detects that the test only covered 35% of the code. And It suggests to used `vapply` instead of `sapply`in `tests/testthat/test-transmute.R`.
- There are many English typos in the documentation (e.g. accross, aditional, compositon, etc). You can see the words to fix [here](https://github.com/quishqa/my_ropensci_reviews/blob/main/rb3-review/rb3-review.md#spell-check).
- When running `test()` there is a warning about [tidyselect](https://github.com/quishqa/my_ropensci_reviews/blob/main/rb3-review/rb3-review.md#run-tests-on-rb3-source): `Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0`.
You can find more information about this warning [here](https://github.com/tidyverse/tidyverse.org/pull/600) and [here](https://community.rstudio.com/t/use-of-data-in-tidyselect-expressions-is-now-deprecated/150092).
- I ran the [Equality data example](https://github.com/wilsonfreitas/rb3#equity-data) without implementing ssl error fix  perfectly fine. I use Ubuntu, maybe it happens in another distro. The test `"it should download a file with a datetime downloader` also ran without problems.
- Also I think it's a good idea to implement GitHub action to regularly check the package.

Congrats for this nice package  !  👍🏽

**Comment by wilsonfreitas** [comment]
Hi @pachadotdev and @quishqa,

I appreciate your suggestions.
I'll work in the next weeks to get them implemented.
Specially the test coverage for downloads.

Bests,

**Comment by emilyriederer** [comment]
Hi @wilsonfreitas ! I hope you're well. I just thought I'd check in and see how your updates are going? In general, we try to aim to get an update to our reviewers in ~2-3 weeks after their reviews so they can evaluate any changes while the information is still somewhat fresh in their minds. Thanks!

**Comment by emilyriederer** [comment]
@ropensci-review-bot submit review https://github.com/ropensci/software-review/issues/534#issuecomment-1295555685 time 6

**Comment by emilyriederer** [comment]
@ropensci-review-bot submit review https://github.com/ropensci/software-review/issues/534#issuecomment-1289832660 time 3

**Comment by wilsonfreitas** [comment]
Hi @emilyriederer,

I've just pushed changed suggested by @quishqa .

- English typos correction
- use vapply instead of sapply
- removed .data where it is deprecated

I am working now on tests to improve coverage with vcr, as @pachadotdev suggested.

**Comment by pachadotdev** [comment]
> Hi @pachadotdev,
> 
> I added vcr to the tests. I'm not finished yet. I'd appreciate it if you could give me any suggestions, especially to understand that I'm on the correct path.
> 
> This [commit](76c13e795fb5d5c2c27550bdd0f63341911cf476) has the vcr implementation.
> 
> I believe I'll have it finished this weekend.
> 
> thank you so much for your attention and participation.

Awesome!!! I have to present thesis progress this week, so I'll be quite disconnected. After that, I can give a careful look.

Right now, I am redeploying my server, because I have over 700 GB of data, and I needed a cheaper VPS.

Best,

**Comment by wilsonfreitas** [comment]
Great @pachadotdev,

I've implemented more tests and I believe we can go on with the review process.

**Comment by emilyriederer** [comment]
Hi @wilsonfreitas ! Thanks for the heads up. Just to confirm: does that mean you're ready for the reviewers to look again for their final approval?

**Comment by wilsonfreitas** [comment]
Hi @emilyriederer,
Yes, I am.

**Comment by emilyriederer** [comment]
Hi there @pachadotdev and @quishqa ! Could you please take a look an @wilsonfreitas 's changes and complete the [approval template](https://devguide.ropensci.org/reviewerguide.html#followupreviewer)? Thank you!

**Comment by emilyriederer** [comment]
Hi @pachadotdev and @quishqa - happy new year! I know my last message might have been poorly timed around the holidays. Would you have time soon to do a final review and approval of @wilsonfreitas 's updates? Thank you!

**Comment by quishqa** [comment]
Hi all! 
I believe the commits addressed my suggestions, but it seems that [R-CMD-Check is failing](https://github.com/wilsonfreitas/rb3/actions/runs/3664411763).

**Comment by wilsonfreitas** [comment]
Hi @quishqa, I forgot to update the GitHub workflows.
The bizdays package has also been updated, and it is a dependency.
It's submitted to CRAN, I'll wait until it's approved to rerun the workflows.

**Comment by pachadotdev** [comment]
> Hi @pachadotdev and @quishqa - happy new year! I know my last message might have been poorly timed around the holidays. Would you have time soon to do a final review and approval of @wilsonfreitas 's updates? Thank you!

hi!
looks good!

**Comment by emilyriederer** [comment]
Thanks for checking in @quishqa and @pachadotdev ! Once @wilsonfreitas has the checks back, can you both please formally sign off with the [template](https://devguide.ropensci.org/reviewerguide.html#followupreviewer)?

**Comment by wilsonfreitas** [comment]
Hi @pachadotdev and @quishqa, all [RMD checks](https://github.com/wilsonfreitas/rb3/actions/runs/3990979812) are passing.

**Comment by quishqa** [comment]
> Thanks for checking in @quishqa and @pachadotdev ! Once @wilsonfreitas has the checks back, can you both please formally sign off with the [template](https://devguide.ropensci.org/reviewerguide.html#followupreviewer)?

Just double-check [this is the template](https://devguide.ropensci.org/approval2template.html), right?

**Comment by emilyriederer** [comment]
@quishqa - Yes, perfect! Thanks. (cc-ing @pachadotdev as well)

**Comment by quishqa** [comment]
## Reviewer Response


#### Final approval (post-review)

- [x] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**



Estimated hours spent reviewing: 7

**Comment by pachadotdev** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below.  Your review is not limited to these topics, as described in the reviewer guide*

- **Briefly describe any working relationship you have (had) with the package authors.**
- [x] As the reviewer I confirm that there are no [conflicts of interest](https://devguide.ropensci.org/policies.html#coi) for me to review this work (if you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need:** clearly stating problems the software is designed to solve and its target audience in README
- [x] **Installation instructions:** for the development version of package and any non-standard dependencies in README
- [x] **Vignette(s):** demonstrating major functionality that runs successfully locally
- [x] **Function Documentation:** for all exported functions
- [x] **Examples:** (that run successfully locally) for all exported functions
- [x] **Community guidelines:** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:** Unit tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines.

Estimated hours spent reviewing: 3

- [x] Should the author(s) deem it appropriate, I agree to be acknowledged as a package reviewer ("rev" role) in the package DESCRIPTION file.

---

### Review Comments

Good!

**Comment by emilyriederer** [comment]
@ropensci-review-bot approve rb3

**Comment by emilyriederer** [comment]
Congratulations, @wilsonfreitas ! Yuur package has been approved. There's a lot of great information above about how you can fully transfer the package to rOpenSci and some (optional) ideas for marketing. 

Would you also be interested in writing a blog post about your package? I think it's always great to feature new data packages in posts since it can raise awareness both for the package and for new underlying APIs that folks may not know about. Additionally, I think there are a lot of synergies between this package and other @msperlin and @pachadotdev submissions, so we could even explore those interactions. If so, @ropensci/blog-editors can provide more detail.

**Comment by maelle** [comment]
@wilsonfreitas @msperlin did you meet any difficulty? I'm asking as I'm drafting rOpenSci newsletter I'll send on Friday: if your package is transferred by then, it will be mentioned in this month's newsletter (if not, in next month's so no worries).

**Comment by msperlin** [comment]
Thanks @maelle. I belive only @wilsonfreitas  can change the ownership of the repository.

**Comment by wilsonfreitas** [comment]
@ropensci-review-bot invite me to ropensci/rb3

**Comment by wilsonfreitas** [comment]
hi @emilyriederer 

I am trying to transfer the repo to ropensci, but I don't have the permission.

![image](https://user-images.githubusercontent.com/1583922/219600754-ec596034-144b-4bf6-af20-91b572c786c9.png)

Am I doing it correctly?

**Comment by maelle** [comment]
@wilsonfreitas I've now invited you to the GitHub organization, can you please try again?

I've just realized this submission is a scenario we hadn't thought of: the submitter @msperlin didn't own the repository, and we send the organization invitation to the submitter. :sweat_smile: Sorry about the resulting clunkiness!

**Comment by maelle** [comment]
In the end the repo was transferred riiight before I sent the newsletter (in a few minutes), well done :trophy: :wink:

**Comment by wilsonfreitas** [comment]
@ropensci-review-bot finalize transfer of ropensci/rb3

**Comment by maelle** [comment]
@wilsonfreitas @msperlin you're now both part of a team with admin access to the repo.
